compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java

```diff
@@ -167,6 +168,15 @@ boolean isDefaultModule() {
             moduleLoadRequests.addAll(docContext.moduleLoadRequests(PackageDependencyScope.DEFAULT));
         }
 
+        // TODO: Move to compiler extension once new Compiler Extension model is introduced
```

Comment:
Should this be added to every module ? I guess it only need to be added to one ?

Reply:
Yeah when building a package, if `observability-included` is set to true, we need to add it to one of the modules in the package.

Shall I move it to PackageResolution class and add it once ?

Reply:
Fixed in f156a5995066afd87a6fe03c1cdce62c78ecd0bf